AGPs/AGP-Comprehensive-Aragon-Security-Review-Program.md

```diff
@@ -0,0 +1,39 @@
+Author: burrrata (@burrrata)
+Status: Stage IV
+Track: Association
+Created: 2019-10-02
+---
+
+# AGP-X: Comprehensive Aragon Security Review Program
+
+## Description of desired Association policy change
+
+This proposal is to expand the [Aragon Security Review](https://github.com/aragon/security-review/) program to include all projects funded by the AA via AGPs, the Nest program, or the Flock program..
+
+Projects would be required to submit a frozen codebase along with documentation detailing the problem the project solves from a high level, a detailed description of how the project works, and what each function in the codebase is supposed to do. At that point the Aragon Security Review team would review the project and, if satisfactory, hand off the project to a security partner. The security partner would then review the project, present findings, and provide suggestions for improvement. The Aragon project would then need to incorporate all major and critical suggestions provided by the security partner before shipping the project to end users and/or receiving any funding (ANT, DAI, or any denomination) related to finishing the project.
+
+## Motivation for changing this Association policy
+
+Recently, when submitting an [Issue in the security review repo](https://github.com/aragon/security-review/issues/12) no one replied. I was also told in various conversations that the security review program was for Flock and not Nest teams. This experience has led me to formalize an AGP stating how and when Aragon funded projects can request a security audit.
+
+Furthermore, in the recent [1Hive Dandelion Orgs Nest grant](https://github.com/aragon/nest/issues/168) we set aside 15k for a security audit. We have yet to receive concrete quotes, but preliminary conversations indicate that this amount is laughably low. The complexity of Aragon applications increases the time and cost of audits. Also, the opacity of contract security in general makes it hard for developers to asses whether or not they're getting a good price and if all the services they need are covered. Most developers don't have experience with security audits. It's something senior solidity devs probably only do a few times. Because of these reasons, it seems better to let a team of dedicated professionals in the Aragon community handle this process vs letting dev teams figure it out as they go. This reduces costs for Aragon as a whole and ensures a consistent level of security for all funded Aragon projects.
+
+Speaking of devs building on Aragon, the Aragon community is now shipping more projects through A1, Autark, Aragon Black, and the Nest program. These apps are amazing and could unlock tons of value, but only if people use them…
+
+People are wary of DAOs because of “the DAO” hack. For people to trust (and thus use) Aragon we need to go above and beyond to prove that the Aragon platform, and all major releases of Aragon apps, are secure. This is easier said than done.
```

Comment:
```suggestion
People are wary of DAOs because of “The DAO” hack. For people to trust (and thus use) Aragon we need to go above and beyond to prove that the Aragon platform, and all major releases of Aragon apps, are secure. This is easier said than done.
```